### DISCUSSION TITLE
Docker workshop 2.0

### DISCUSSION THREAD

**Discussion by MilesChou** [ganhuaking/meetups#2] [許願池]
使用此書做練習

https://dockerbook.tw/

會上 [Discord](https://discord.io/ganhuaking) 上做線上直播討論。

預計會有約 3~4 次，看講的進度來決定

* [x] 11/4 22:00
* [x] 11/11 22:00
* [x] 11/18 22:00
* [ ] 未定 22:00 docker build

**Comment by yuanyu90221** [comment]
感謝師傅

**Comment by MilesChou** [comment]
11/4 場順利結束，感謝大家

這次分享約一小時，感覺控制在一小時結束應該很ok。這次講的內容如下：

1. docker run hello-world 詳解
2. 介紹 port forwarding

**Comment by MilesChou** [comment]
Dockercraft

https://github.com/docker/dockercraft

後來有研究了一下，因為當時使用 Lima ，裡面執行的 user 不是 root ，所以才會出現權限錯誤：

```
FATA[0000] Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.30/info: dial unix /var/run/docker.sock: connect: permission denied
```
後來是進 Lima VM，改用 sudo 解決。

後續出現的 Connection refused 還不確定是什麼原因，但推測是 docker.sock 權限有問題的關係，所以這段連線會一直被拒絕。改 sudo 和加上 --privileged 都不是完全有效（時好時壞），因此還不確定發生什麼問題。

**Comment by MilesChou** [comment]
11/11 記錄

如果把兩個 container 放在同個 network 時，想要做成單向連結時，只能靠 iptables，官網有，感謝 @tedmax100 提供

https://docs.docker.com/network/iptables/

**Comment by MilesChou** [comment]
11/18 記錄

使用 Lima 跑 Dockercraft 的話，會需要權限存取 host 的 docker.sock 檔，因此需要進入 Lima 並 sudo 成 root 權限才能正常執行。
